File: sdks/node/e2e/crasher.mjs
// Crasher subprocess for the Node SDK end-to-end proof.
//
// Imports the BUILT bundle (dist/index.mjs, produced by `npm run build`), inits
// the real SDK with a DSN from the environment, then throws an UNCAUGHT error on
// a later tick. The SDK's process-level 'uncaughtException' handler (see
// src/instrument.ts) captures it at level fatal, flushes, and then mirrors
// Node's default by exiting with code 1 - so the event reaches the ingest
// endpoint even though the process crashes.
//
// Run by run_e2e.mjs, never directly. Reads CRASHLENS_DSN from the environment.

import { init } from "../dist/index.mjs";
import { MARKER } from "./marker.mjs";

init({
  dsn: process.env.CRASHLENS_DSN,
  release: "nodesvc@1.0.0",
  environment: "production",
});

// Uncaught on purpose, deferred one tick so the handler is installed first.
setImmediate(() => {
  throw new Error(MARKER);
});

File: sdks/python/e2e/crasher.py
"""Crasher subprocess for the Python SDK end-to-end proof.

Initialises the real Crashlens Python SDK with a DSN taken from the environment
and then raises an UNCAUGHT exception. The SDK's installed ``sys.excepthook``
(see crashlens/_hooks.py) captures the exception at level ``fatal`` and flushes
the queue synchronously before the interpreter exits, so the event reaches the
ingest endpoint even though the process terminates with a non-zero status.

Run by e2e/run_e2e.py, never directly. Reads:
  CRASHLENS_DSN - the full DSN (http://<key>@host/api/ingest/<project>/).
"""

import os

import crashlens

MARKER = "crashlens python e2e uncaught marker"


def main() -> None:
    dsn = os.environ["CRASHLENS_DSN"]
    crashlens.init(dsn, release="pysvc@1.0.0", environment="production")
    # Uncaught on purpose: the excepthook path is what we are proving.
    raise RuntimeError(MARKER)


if __name__ == "__main__":
    main()
